### ISSUE TITLE
[Issue]: Alphanumeric sorting of songs separates uppercase from lowercase letters

### ISSUE THREAD

**Issue by Chaphasilor** [jellyfin/jellyfin#11320]
### Please describe your bug

When using the "Track Name" sort order on the "Songs" tab within a music library, it starts by listing all songs starting with an uppercase letter from A-Z, and then continues with songs starting with lowercase letters from a-z.  

Usually, this is not how alphanumeric sorting behaves. Instead, the cases (casings?) would be separated on a per-letter basis only, not globally.

### Reproduction Steps

1. Go to a music library with songs starting with uppercase and lowercase letters
2. Go to the songs tab
3. Make sure sorting is set to "Track Name"
4. Scroll to the last song starting with an uppercase letter
5. The next song starts with a lowercase letter

### Jellyfin Version

Unstable (master branch)

### if other:

10.8.13 is also affected

### Environment

```markdown
- OS: TrueNAS Scale, based on Debian 11
- Linux Kernel:
- Virtualization: Docker
- Clients: Web, Finamp
- Browser: Edge
- FFmpeg Version:
- Playback Method:
- Hardware Acceleration:
- GPU Model:
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
n/a
```

### Please attach any screenshots here

![image](https://github.com/jellyfin/jellyfin/assets/18015147/ffb375ab-4878-4008-9657-abd4d5b09dce)

(to reduce scrolling, I reversed the sort order, so it starts with lowercase)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Chaphasilor** [comment]
@jellyfin-bot fight me

**Comment by felix920506** [comment]
I think the behavior really depends on the implementation. I've seen places where the case gets ignored but I've also seen places where upper and lower case are completely separated.

**Comment by Chaphasilor** [comment]
Yeah so it seems like for all other lists (albums, artists, genres, etc.) it works as intended, just for songs it's different.  
The songs list also looks different visually from the other lists.

**Comment by storfors** [comment]
I don't believe this issue was fixed? The first commit of PR #14153 did fix it, but the second commit just changed the name ordering from "Name" to "CleanName" and reintroduced the case sensitive ordering.

**Comment by gnattu** [comment]
The cleaname is case normalized.
